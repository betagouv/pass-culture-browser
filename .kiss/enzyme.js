import React from 'react'
import { shallow } from 'enzyme'

import MyComponent from '../MyComponent'

describe('src | components | MyComponent', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      const props = { errors: {} }
      const wrapper = shallow(<MyComponent {...props} />)
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
