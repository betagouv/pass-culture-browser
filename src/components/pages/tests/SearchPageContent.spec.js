import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import React from 'react'
import { shallow } from 'enzyme'

import SearchPageContent from '../SearchPageContent'

const middlewares = []
const mockStore = configureStore(middlewares)

describe('src | components | pages | SearchPageContentContent', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      // given
      const initialState = {}
      const store = mockStore(initialState)
      const props = {
        dispatch: jest.fn(),
        history: {},
        location: {},
        match: {},
        pagination: {
          windowQuery: {
            categories: null,
            date: null,
            distance: null,
            jours: null,
            latitude: null,
            longitude: null,
            'mots-cles': 'Fake',
            orderBy: 'offer.id+desc',
          },
        },
        recommendations: [],
        search: {},
        typeSublabels: [],
        typeSublabelsAndDescription: [],
      }

      // when
      const wrapper = shallow(
        <Provider store={store}>
          <SearchPageContent {...props} />
        </Provider>
      )

      // then
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
  //
  // describe.skip('render', () => {
  //   // given
  //   const initialState = REDUX_STATE
  //   const store = mockStore(initialState)
  //   const wrapper = shallow(<SearchPageContent store={store} />)
  //
  // })

  describe('functions', () => {
    describe('constructor', () => {
      it('should initialize state correctly', () => {
        // given
        const props = {
          dispatch: jest.fn(),
          history: {},
          location: {
            hash: '',
            key: 'lxn6vp',
            pathname: '/recherche',
            search: '?orderBy=offer.id+desc',
            state: undefined,
          },
          match: {
            params: {
              categorie: undefined,
              view: undefined,
            },
          },
          pagination: {
            windowQuery: {
              categories: null,
              date: null,
              distance: null,
              jours: null,
              latitude: null,
              longitude: null,
              'mots-cles': 'Fake',
              orderBy: 'offer.id+desc',
            },
          },
          recommendations: [],
          search: {},
          typeSublabels: [],
          typeSublabelsAndDescription: [],
        }

        // when
        const wrapper = shallow(<SearchPageContent {...props} />)
        const expected = {
          keywordsKey: 0,
          keywordsValue: 'Fake',
          withFilter: false,
        }

        // then
        expect(wrapper.state()).toEqual(expected)
      })
    })
    describe('handleDataRequest', () => {
      it('should', () => {})
    })
    describe('loadMoreHandler', () => {
      it('should', () => {})
    })
    describe('onBackToSearchHome', () => {
      it('should', () => {})
    })
    describe('onSubmit', () => {
      it('should', () => {})
    })
    describe('onClickOpenCloseFilterDiv', () => {
      it('should', () => {})
    })
    describe('onKeywordsChange', () => {
      it('should', () => {})
    })
    describe('onKeywordsEraseClick', () => {
      it('should', () => {})
    })
  })
})

// Titre de la page selon s'il y a des recommendations ou pas.
