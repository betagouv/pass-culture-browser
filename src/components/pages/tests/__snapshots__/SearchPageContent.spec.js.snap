// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | SearchPageContent snapshot should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchPageContent
    dispatch={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "patch": Object {
                "recommendations": Array [],
              },
              "type": "ASSIGN_DATA",
            },
          ],
          Array [
            Object {
              "config": Object {},
              "method": "GET",
              "path": "types",
              "type": "REQUEST_DATA_GET_TYPES",
            },
          ],
          Array [
            Object {
              "patch": Object {
                "recommendations": Array [],
              },
              "type": "ASSIGN_DATA",
            },
          ],
          Array [
            Object {
              "config": Object {},
              "method": "GET",
              "path": "types",
              "type": "REQUEST_DATA_GET_TYPES",
            },
          ],
          Array [
            Object {
              "patch": Object {
                "recommendations": Array [],
              },
              "type": "ASSIGN_DATA",
            },
          ],
          Array [
            Object {
              "config": Object {},
              "method": "GET",
              "path": "types",
              "type": "REQUEST_DATA_GET_TYPES",
            },
          ],
          Array [
            Object {
              "patch": Object {
                "recommendations": Array [],
              },
              "type": "ASSIGN_DATA",
            },
          ],
          Array [
            Object {
              "config": Object {},
              "method": "GET",
              "path": "types",
              "type": "REQUEST_DATA_GET_TYPES",
            },
          ],
          Array [
            Object {
              "patch": Object {
                "recommendations": Array [],
              },
              "type": "ASSIGN_DATA",
            },
          ],
          Array [
            Object {
              "config": Object {},
              "method": "GET",
              "path": "types",
              "type": "REQUEST_DATA_GET_TYPES",
            },
          ],
          Array [
            Object {
              "patch": Object {
                "recommendations": Array [],
              },
              "type": "ASSIGN_DATA",
            },
          ],
          Array [
            Object {
              "config": Object {},
              "method": "GET",
              "path": "types",
              "type": "REQUEST_DATA_GET_TYPES",
            },
          ],
          Array [
            Object {
              "patch": Object {
                "recommendations": Array [],
              },
              "type": "ASSIGN_DATA",
            },
          ],
          Array [
            Object {
              "config": Object {},
              "method": "GET",
              "path": "types",
              "type": "REQUEST_DATA_GET_TYPES",
            },
          ],
          Array [
            Object {
              "patch": Object {
                "recommendations": Array [],
              },
              "type": "ASSIGN_DATA",
            },
          ],
          Array [
            Object {
              "config": Object {},
              "method": "GET",
              "path": "types",
              "type": "REQUEST_DATA_GET_TYPES",
            },
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    history={
      Object {
        "push": [MockFunction],
      }
    }
    location={
      Object {
        "hash": "",
        "key": "lxn6vp",
        "pathname": "/recherche",
        "search": "?orderBy=offer.id+desc",
        "state": undefined,
      }
    }
    match={
      Object {
        "params": Object {
          "categorie": undefined,
          "view": "resultats",
        },
      }
    }
    query={
      Object {
        "change": [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "page": null,
              },
            ],
            Array [
              Object {
                "page": null,
              },
            ],
            Array [
              Object {
                "page": null,
              },
            ],
            Array [
              Object {
                "page": null,
              },
            ],
            Array [
              Object {
                "page": null,
              },
            ],
            Array [
              Object {
                "page": null,
              },
            ],
            Array [
              Object {
                "page": null,
              },
            ],
            Array [
              Object {
                "page": null,
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
        "parse": [Function],
      }
    }
    recommendations={Array []}
    search={Object {}}
    typeSublabels={Array []}
    typeSublabelsAndDescription={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "backButton": Object {
        "onClick": [Function],
      },
      "children": Array [
        <form
          onSubmit={[Function]}
        >
          <div
            className="flex-columns items-start"
          >
            <div
              className="field has-addons flex-columns flex-1"
              id="search-page-keywords-field"
            >
              <p
                className="control has-icons-right flex-1"
              >
                <input
                  className="input search-input"
                  id="keywords"
                  onChange={[Function]}
                  placeholder="Un mot-clÃ©"
                  type="text"
                />
              </p>
              <div
                className="control flex-0"
              >
                <button
                  className="button is-rounded is-medium"
                  disabled={true}
                  id="keywords-search-button"
                  type="submit"
                >
                  Chercher
                </button>
              </div>
            </div>
            <div
              className="flex-0 text-center flex-rows flex-center pb12 "
              id="search-filter-menu-toggle-button"
            >
              <button
                className="no-border no-background no-outline "
                onClick={[Function]}
                type="button"
              >
                <Icon
                  svg="ico-filter-active"
                />
              </button>
            </div>
          </div>
        </form>,
        <withRouter(_withQueryRouter)
          isVisible={false}
        />,
        <Switch
          location={
            Object {
              "hash": "",
              "key": "lxn6vp",
              "pathname": "/recherche",
              "search": "?orderBy=offer.id+desc",
              "state": undefined,
            }
          }
        >
          <Route
            exact={true}
            path="/recherche"
            render={[Function]}
          />
          <Route
            path="/recherche/resultats/:categorie"
            render={[Function]}
          />
          <Route
            path="/recherche/resultats"
            render={[Function]}
          />
        </Switch>,
      ],
      "closeSearchButton": true,
      "footer": [Function],
      "handleDataRequest": [Function],
      "header": [Function],
      "id": "search-page",
      "name": "search",
      "pageTitle": "Recherche : rÃ©sultats",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="flex-columns items-start"
          >
            <div
              className="field has-addons flex-columns flex-1"
              id="search-page-keywords-field"
            >
              <p
                className="control has-icons-right flex-1"
              >
                <input
                  className="input search-input"
                  id="keywords"
                  onChange={[Function]}
                  placeholder="Un mot-clÃ©"
                  type="text"
                />
              </p>
              <div
                className="control flex-0"
              >
                <button
                  className="button is-rounded is-medium"
                  disabled={true}
                  id="keywords-search-button"
                  type="submit"
                >
                  Chercher
                </button>
              </div>
            </div>
            <div
              className="flex-0 text-center flex-rows flex-center pb12 "
              id="search-filter-menu-toggle-button"
            >
              <button
                className="no-border no-background no-outline "
                onClick={[Function]}
                type="button"
              >
                <Icon
                  svg="ico-filter-active"
                />
              </button>
            </div>
          </div>,
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="field has-addons flex-columns flex-1"
                id="search-page-keywords-field"
              >
                <p
                  className="control has-icons-right flex-1"
                >
                  <input
                    className="input search-input"
                    id="keywords"
                    onChange={[Function]}
                    placeholder="Un mot-clÃ©"
                    type="text"
                  />
                </p>
                <div
                  className="control flex-0"
                >
                  <button
                    className="button is-rounded is-medium"
                    disabled={true}
                    id="keywords-search-button"
                    type="submit"
                  >
                    Chercher
                  </button>
                </div>
              </div>,
              <div
                className="flex-0 text-center flex-rows flex-center pb12 "
                id="search-filter-menu-toggle-button"
              >
                <button
                  className="no-border no-background no-outline "
                  onClick={[Function]}
                  type="button"
                >
                  <Icon
                    svg="ico-filter-active"
                  />
                </button>
              </div>,
            ],
            "className": "flex-columns items-start",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="control has-icons-right flex-1"
                  >
                    <input
                      className="input search-input"
                      id="keywords"
                      onChange={[Function]}
                      placeholder="Un mot-clÃ©"
                      type="text"
                    />
                  </p>,
                  <div
                    className="control flex-0"
                  >
                    <button
                      className="button is-rounded is-medium"
                      disabled={true}
                      id="keywords-search-button"
                      type="submit"
                    >
                      Chercher
                    </button>
                  </div>,
                ],
                "className": "field has-addons flex-columns flex-1",
                "id": "search-page-keywords-field",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <input
                        className="input search-input"
                        id="keywords"
                        onChange={[Function]}
                        placeholder="Un mot-clÃ©"
                        type="text"
                      />,
                      undefined,
                    ],
                    "className": "control has-icons-right flex-1",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "input search-input",
                        "defaultValue": undefined,
                        "id": "keywords",
                        "onChange": [Function],
                        "placeholder": "Un mot-clÃ©",
                        "type": "text",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                    undefined,
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <button
                      className="button is-rounded is-medium"
                      disabled={true}
                      id="keywords-search-button"
                      type="submit"
                    >
                      Chercher
                    </button>,
                    "className": "control flex-0",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Chercher",
                      "className": "button is-rounded is-medium",
                      "disabled": true,
                      "id": "keywords-search-button",
                      "type": "submit",
                    },
                    "ref": null,
                    "rendered": "Chercher",
                    "type": "button",
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="no-border no-background no-outline "
                  onClick={[Function]}
                  type="button"
                >
                  <Icon
                    svg="ico-filter-active"
                  />
                </button>,
                "className": "flex-0 text-center flex-rows flex-center pb12 ",
                "id": "search-filter-menu-toggle-button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Icon
                    svg="ico-filter-active"
                  />,
                  "className": "no-border no-background no-outline ",
                  "onClick": [Function],
                  "type": "button",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "svg": "ico-filter-active",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "button",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "isVisible": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              exact={true}
              path="/recherche"
              render={[Function]}
            />,
            <Route
              path="/recherche/resultats/:categorie"
              render={[Function]}
            />,
            <Route
              path="/recherche/resultats"
              render={[Function]}
            />,
          ],
          "location": Object {
            "hash": "",
            "key": "lxn6vp",
            "pathname": "/recherche",
            "search": "?orderBy=offer.id+desc",
            "state": undefined,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/recherche",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/recherche/resultats/:categorie",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/recherche/resultats",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "backButton": Object {
          "onClick": [Function],
        },
        "children": Array [
          <form
            onSubmit={[Function]}
          >
            <div
              className="flex-columns items-start"
            >
              <div
                className="field has-addons flex-columns flex-1"
                id="search-page-keywords-field"
              >
                <p
                  className="control has-icons-right flex-1"
                >
                  <input
                    className="input search-input"
                    id="keywords"
                    onChange={[Function]}
                    placeholder="Un mot-clÃ©"
                    type="text"
                  />
                </p>
                <div
                  className="control flex-0"
                >
                  <button
                    className="button is-rounded is-medium"
                    disabled={true}
                    id="keywords-search-button"
                    type="submit"
                  >
                    Chercher
                  </button>
                </div>
              </div>
              <div
                className="flex-0 text-center flex-rows flex-center pb12 "
                id="search-filter-menu-toggle-button"
              >
                <button
                  className="no-border no-background no-outline "
                  onClick={[Function]}
                  type="button"
                >
                  <Icon
                    svg="ico-filter-active"
                  />
                </button>
              </div>
            </div>
          </form>,
          <withRouter(_withQueryRouter)
            isVisible={false}
          />,
          <Switch
            location={
              Object {
                "hash": "",
                "key": "lxn6vp",
                "pathname": "/recherche",
                "search": "?orderBy=offer.id+desc",
                "state": undefined,
              }
            }
          >
            <Route
              exact={true}
              path="/recherche"
              render={[Function]}
            />
            <Route
              path="/recherche/resultats/:categorie"
              render={[Function]}
            />
            <Route
              path="/recherche/resultats"
              render={[Function]}
            />
          </Switch>,
        ],
        "closeSearchButton": true,
        "footer": [Function],
        "handleDataRequest": [Function],
        "header": [Function],
        "id": "search-page",
        "name": "search",
        "pageTitle": "Recherche : rÃ©sultats",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="flex-columns items-start"
            >
              <div
                className="field has-addons flex-columns flex-1"
                id="search-page-keywords-field"
              >
                <p
                  className="control has-icons-right flex-1"
                >
                  <input
                    className="input search-input"
                    id="keywords"
                    onChange={[Function]}
                    placeholder="Un mot-clÃ©"
                    type="text"
                  />
                </p>
                <div
                  className="control flex-0"
                >
                  <button
                    className="button is-rounded is-medium"
                    disabled={true}
                    id="keywords-search-button"
                    type="submit"
                  >
                    Chercher
                  </button>
                </div>
              </div>
              <div
                className="flex-0 text-center flex-rows flex-center pb12 "
                id="search-filter-menu-toggle-button"
              >
                <button
                  className="no-border no-background no-outline "
                  onClick={[Function]}
                  type="button"
                >
                  <Icon
                    svg="ico-filter-active"
                  />
                </button>
              </div>
            </div>,
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="field has-addons flex-columns flex-1"
                  id="search-page-keywords-field"
                >
                  <p
                    className="control has-icons-right flex-1"
                  >
                    <input
                      className="input search-input"
                      id="keywords"
                      onChange={[Function]}
                      placeholder="Un mot-clÃ©"
                      type="text"
                    />
                  </p>
                  <div
                    className="control flex-0"
                  >
                    <button
                      className="button is-rounded is-medium"
                      disabled={true}
                      id="keywords-search-button"
                      type="submit"
                    >
                      Chercher
                    </button>
                  </div>
                </div>,
                <div
                  className="flex-0 text-center flex-rows flex-center pb12 "
                  id="search-filter-menu-toggle-button"
                >
                  <button
                    className="no-border no-background no-outline "
                    onClick={[Function]}
                    type="button"
                  >
                    <Icon
                      svg="ico-filter-active"
                    />
                  </button>
                </div>,
              ],
              "className": "flex-columns items-start",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p
                      className="control has-icons-right flex-1"
                    >
                      <input
                        className="input search-input"
                        id="keywords"
                        onChange={[Function]}
                        placeholder="Un mot-clÃ©"
                        type="text"
                      />
                    </p>,
                    <div
                      className="control flex-0"
                    >
                      <button
                        className="button is-rounded is-medium"
                        disabled={true}
                        id="keywords-search-button"
                        type="submit"
                      >
                        Chercher
                      </button>
                    </div>,
                  ],
                  "className": "field has-addons flex-columns flex-1",
                  "id": "search-page-keywords-field",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": "0",
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <input
                          className="input search-input"
                          id="keywords"
                          onChange={[Function]}
                          placeholder="Un mot-clÃ©"
                          type="text"
                        />,
                        undefined,
                      ],
                      "className": "control has-icons-right flex-1",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "input search-input",
                          "defaultValue": undefined,
                          "id": "keywords",
                          "onChange": [Function],
                          "placeholder": "Un mot-clÃ©",
                          "type": "text",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "input",
                      },
                      undefined,
                    ],
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <button
                        className="button is-rounded is-medium"
                        disabled={true}
                        id="keywords-search-button"
                        type="submit"
                      >
                        Chercher
                      </button>,
                      "className": "control flex-0",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Chercher",
                        "className": "button is-rounded is-medium",
                        "disabled": true,
                        "id": "keywords-search-button",
                        "type": "submit",
                      },
                      "ref": null,
                      "rendered": "Chercher",
                      "type": "button",
                    },
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <button
                    className="no-border no-background no-outline "
                    onClick={[Function]}
                    type="button"
                  >
                    <Icon
                      svg="ico-filter-active"
                    />
                  </button>,
                  "className": "flex-0 text-center flex-rows flex-center pb12 ",
                  "id": "search-filter-menu-toggle-button",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Icon
                      svg="ico-filter-active"
                    />,
                    "className": "no-border no-background no-outline ",
                    "onClick": [Function],
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "svg": "ico-filter-active",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "button",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "isVisible": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                exact={true}
                path="/recherche"
                render={[Function]}
              />,
              <Route
                path="/recherche/resultats/:categorie"
                render={[Function]}
              />,
              <Route
                path="/recherche/resultats"
                render={[Function]}
              />,
            ],
            "location": Object {
              "hash": "",
              "key": "lxn6vp",
              "pathname": "/recherche",
              "search": "?orderBy=offer.id+desc",
              "state": undefined,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/recherche",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/recherche/resultats/:categorie",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/recherche/resultats",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
